src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -900,59 +1034,50 @@ public async Task SslStream_ClientCertificate_SendsChain()
+            {
+                TargetHost = "localhost",
+                EnabledSslProtocols = SslProtocols.Tls12
+            };
             clientOptions.RemoteCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true;
             clientOptions.LocalCertificateSelectionCallback = (sender, target, certificates, remoteCertificate, issuers) => clientCertificate;
 
-            var serverOptions = new SslServerAuthenticationOptions() { ClientCertificateRequired = true };
-            serverOptions.ServerCertificateContext = SslStreamCertificateContext.Create(Configuration.Certificates.GetServerCertificate(), null);
-            serverOptions.RemoteCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) =>
+            await SslStream_ClientSendsChain_Core(clientOptions, clientChain);
+
+            TestHelper.CleanupCertificates(nameof(SslStream_ClientCertificate_SendsChain), storeName);
+            clientCertificate.Dispose();
+            foreach (X509Certificate c in clientChain)
             {
-                // Client should send chain without root CA. There is no good way how to know if the chain was built from certificates
-                // from wire or from system store. However, SslStream adds certificates from wire to ExtraStore in RemoteCertificateValidationCallback.
-                // So we verify the operation by checking the ExtraStore. On Windows, that includes leaf itself.
-                _output.WriteLine("RemoteCertificateValidationCallback called with {0} and {1} extra certificates", sslPolicyErrors, chain.ChainPolicy.ExtraStore.Count);
-                foreach (X509Certificate c in chain.ChainPolicy.ExtraStore)
-                {
-                    _output.WriteLine("received {0}", c.Subject);
-                }
+                c.Dispose();
+            }
+        }
 
-                // We may get completed chain from OS even if client sent less.
-                // As minimum, it should send more than the leaf cert
-                Assert.True(chain.ChainPolicy.ExtraStore.Count >= clientChain.Count - 1);
-                Assert.Contains(clientChain[0], chain.ChainPolicy.ExtraStore);
-                return true;
-            };
+        [Fact]
+        [SkipOnPlatform(TestPlatforms.Android, "Self-signed certificates are rejected by Android before the .NET validation is reached")]
+        public async Task SslStream_ClientCertificateContext_SendsChain()
+        {
+            (X509Certificate2 clientCertificate, X509Certificate2Collection clientChain) = TestHelper.GenerateCertificates(nameof(SslStream_ClientCertificateContext_SendsChain), serverCertificate: false);
+            TestHelper.CleanupCertificates(nameof(SslStream_ClientCertificateContext_SendsChain));
 
-            // run the test multiple times while holding established SSL so we could hit credential cache.
-            for (int i = 0; i < 3; i++)
```

Comment:
Is there reason why this was removed? We may have better coverage now but we had problems in the past when first and second try gives different results for same inputs.

Reply:
ah.  you move it to the core function....